### ISSUE TITLE
lerna version/publish creates duplicate versions + extra whitespace in change log

### ISSUE THREAD

**Issue by jedmao** [lerna/lerna#2116]
Every Git commit by Lerna after `lerna publish` has a duplicate version in the change log, each with 4 extra newlines that follow.

## Expected Behavior
- Notes for a particular version should only appear once in a change log.
- No additional whitespace under each version.

## Current Behavior

<details><summary>CHANGELOG.md</summary><p>

```md
## [0.1.1](https://github.com/adobe/generator-tsx/compare/v0.1.0...v0.1.1) (2019-05-29)

**Note:** Version bump only for package generator-tsx





## [0.1.1](https://github.com/adobe/generator-tsx/compare/v0.1.0...v0.1.1) (2019-05-29)

**Note:** Version bump only for package generator-tsx





```
</p></details>

<details><summary>Travis-CI Build</summary><p>

```bash
> generator-tsx@0.1.0 release /home/travis/build/adobe/generator-tsx
> cross-env HUSKY_BYPASS=true lerna publish "--yes" "--git-remote" "pub"
lerna notice cli v3.14.1
lerna info ci enabled
lerna info publish rooted leaf detected, skipping synthetic root lifecycles
lerna info current version 0.1.0
lerna info Looking for changed packages since v0.1.0
lerna info getChangelogConfig Successfully resolved preset "conventional-changelog-angular"
Changes:
 - generator-tsx: 0.1.0 => 0.1.1
lerna info auto-confirmed 
lerna info execute Skipping GitHub releases
lerna info git Pushing tags...
lerna info publish Publishing packages to npm...
lerna info Verifying npm credentials
lerna http fetch GET 200 https://registry.npmjs.org/-/npm/v1/user 545ms
lerna http fetch GET 200 https://registry.npmjs.org/-/org/jedmao/package?format=cli 131ms
lerna info lifecycle generator-tsx@0.1.1~prepublishOnly: generator-tsx@0.1.1
lerna success published generator-tsx 0.1.1
lerna notice 
lerna notice 📦  generator-tsx@0.1.1
lerna notice === Tarball Contents === 
lerna notice 4.3kB  package.json                                                                         
lerna notice 1.1kB  CHANGELOG.md                                                                         
lerna notice 1.1kB  LICENSE                                                                              
lerna notice 4.3kB  README.md                                                                 
lerna notice === Tarball Details === 
lerna notice name:          generator-tsx                           
lerna notice version:       0.1.1                                   
lerna notice filename:      generator-tsx-0.1.1.tgz                 
lerna notice package size:  68.9 kB                                 
lerna notice unpacked size: 208.6 kB                                                               
lerna notice 
lerna http fetch PUT 200 https://registry.npmjs.org/generator-tsx 3646ms
Successfully published:
 - generator-tsx@0.1.1
lerna success published 1 package
Done. Your build exited with 0.
```
</p></details>

## Possible Solution
No idea how to fix this, but it may be an external bug with conventional commits?

## Steps to Reproduce (for bugs)

For context, I'm actually using Lerna for a single package.

<details><summary>.travis.yml</summary><p>

```yml
deploy:
  - provider: script
    script: lerna publish --yes --git-remote pub
    skip_cleanup: true
    keep_history: true
    on:
      branch: master
```
</p></details>

<details><summary>lerna.json</summary><p>

```json
{
  "packages": ["."],
  "command": {
    "publish": {
      "allowBranch": ["master"],
      "conventionalCommits": true
    }
  },
  "version": "0.1.1"
}
```
</p></details>

<details><summary>package.json</summary><p>

```json
  "commitlint": {
    "extends": [
      "@commitlint/config-conventional"
    ]
  },
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-conventional-changelog"
    }
  },
```
</p></details>

## Context
I'm automating deploys and change logs, so this is the one thing that's just not working properly at this point.

## Travis-CI Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.14.1 |
| `npm --version`  | 6.9.0 |
| `node --version` | 10.16.0 |

| OS | Version |
| --- | --- |
| Ubuntu | 14.04.5 LTS |

## Project

https://github.com/adobe/generator-tsx

**Comment by revelt** [comment]
Very strange, have a look at https://travis-ci.com/adobe/generator-tsx/builds/113815643#L670 - it's as if release is happening twice... In which case, changelogs will get generated twice...

**Comment by jedmao** [comment]
Yeah the merge commit from the PR is the one that actually does the release (because it's on master) and then Lerna publish creates a new commit, which triggers yet another release. I'm a bit confused about how to handle that.

**Comment by revelt** [comment]
Personally, I stopped using `lerna publish` and went with `lerna version` only. See my recent post https://github.com/lerna/lerna/issues/461#issuecomment-497358557

**Comment by jedmao** [comment]
Actually, that post helped! I think Travis has a way to only publish on a tagged commit. That might make it happen only once!

**Comment by revelt** [comment]
Awesome! Please update if you manage to sort this out, it might be handy for other people.

**Comment by jedmao** [comment]
@revelt, to be clear, the release is only happening once; though, Travis-CI is setup to run `lerna publish` on master, which means the first time it's run, it creates a GitHub commit with the new version and pushes that. Once that gets pushed, it kicks off another `lerna publish`, which fails, which is exactly what I want, because it's already been published.

That said, the build to which you linked me is associated with a single commit that already has a duplicate version in the changelog, so this is not the root of the problem.

I'm still a bit lost here. I think it's a bug.

**Comment by jedmao** [comment]
I just verified that `lerna version` also causes the duplicates.

**Comment by jedmao** [comment]
And now I just verified that lerna is, in fact, responsible for all the extra newlines, but `conventional-changelog-core` is responsible for the duplication.

**Comment by evocateur** [comment]
So, the elephant in the room: Why are you using Lerna for this project?

Tools like [standard-version](https://github.com/conventional-changelog/standard-version#readme) or [semantic-release](https://github.com/semantic-release/semantic-release#readme) are much better for your use-case.

**Comment by jedmao** [comment]
Familiarity. It's what I use in other projects, but those projects don't exhibit this same issue. Rather than dick around with a separate tool until it works, I figured I'd use something I already know how to configure.

That said, I'm happy to switch to semantic-release if it fixes my issues.

**Comment by jedmao** [comment]
- I switched to [`semantic-release`](https://github.com/semantic-release/semantic-release) and introduced a new config called [`@jedmao/semantic-release-npm-github-config`](https://github.com/jedmao/semantic-release-npm-github-config) so that it will be more painless for future projects.
- The changelog issues seem to be resolved now on my end, but this lerna package still has the issue somewhere.
- I guess I won't be using lerna for a single package from now on; though, I don't see any technical reason it shouldn't be able to support that. 🐘 

Closing this issue, only because it seems like an edge case?
